src/algebra/jordan/basic.lean

```diff
@@ -0,0 +1,256 @@
+in `A`,
+```
+(a * b) * a^2 = a * (b * a^2)
+```
+i.e. the operators of multiplication by `a` and `a^2` commute. Every associative algebra can be
+equipped with a second  multiplication making it into a commutative Jordan algebra.
+Jordan algebras arising this way are said to be special. There are also exceptional Jordan algebras
+which can be shown not to be the symmetrisation of any associative algebra. The 3x3
+matrices of octonions is the canonical example.
+
+Commutative Jordan algebras were introduced by Jordan, von Neumann and Wigner
+([jordanvonneumannwigner1934]) as a mathematical model for the observables of a quantum mechanical
+physical system (for a C*-algebra the self-adjoint part is closed under the symmetrised Jordan
+multiplication). Jordan algebras have subsequently been studied from the points of view of abstract
+algebra and functional analysis. They have connections to Lie algebras and differential geometry.
+
+A more general concept of a (non-commutative) Jordan algebra can also be defined, as a
+(non-commutative, non-associative) algebra `A` where, for each `a` in `A`, the operators of left and
+right multiplication by `a` and `a^2` commute. Such algebras have connected to the Vidav-Palmer
+theorem [cabreragarciarodriguezpalacios2014].
+
+A comprehensive overview of the algebraic theory can be found in [mccrimmon2004].
+
+A real Jordan algebra `A` can be introduced by
+```
+variables {A : Type*} [non_unital_non_assoc_ring A] [module ℝ A] [smul_comm_class ℝ A A]
+  [is_scalar_tower ℝ A A] [comm_jordan A]
+```
+
+## Main results
+
+- `lin_jordan` : Linearisation of the commutative Jordan axiom
+
+## Implementation notes
+
+We shall primarily be interested in linear Jordan algebras (i.e. over rings of characteristic not
+two) leaving quadratic algebras to those better versed in that theory.
+
+The conventional way to linearise the Jordan axiom is to equate coefficients (more formally, assume
+that the axiom holds in all field extensions). For simplicity we use brute force algebraic expansion
+and substitution instead.
+
+## References
+
+* [Cabrera García and Rodríguez Palacios, Non-associative normed algebras. Volume 1]
+  [cabreragarciarodriguezpalacios2014]
+* [Hanche-Olsen and Størmer, Jordan Operator Algebras][hancheolsenstormer1984]
+* [Jordan, von Neumann and Wigner, 1934][jordanvonneumannwigner1934]
+* [McCrimmon, A taste of Jordan algebras][mccrimmon2004]
+
+
+-/
+
+set_option old_structure_cmd true
+
+section non_unital_non_assoc_ring
+
+variables {A : Type*} [non_unital_non_assoc_semiring A]
+
+namespace non_unital_non_assoc_ring
+/--
+Left multiplication operator
+-/
+@[simps] def L : A →+ add_monoid.End A := add_monoid_hom.mul
+
+/--
+Right multiplication operator
+-/
+@[simps] def R : A→+(add_monoid.End A) := add_monoid_hom.flip (L : A →+ add_monoid.End A)
+
+lemma L_def (a b : A) : L a b = a*b := rfl
+
+lemma R_def (a b : A) : R a b = b*a := rfl
```

Comment:
How useful is this really, vs the much easier:
```lean
local notation `L` := (*)
local notation `R` := flip (*)
```
Duplication definitions with different names usually isn't that great an idea.

Reply:
Okay, but when `[non_unital_non_assoc_ring A]` I need to find some way to tell Lean that `L` and `R` are endomorphisms in order to be able to write expressions like `⁅L a, R a⁆=0`.

Reply:
Ah, perhaps then these should be called `function.End.R` and `function.End.L`?

Reply:
Done, thanks!